src/main/java/uk/ac/ic/wlgitbridge/bridge/repo/FSGitRepoStore.java

```diff
@@ -22,10 +23,16 @@
 
     private final String repoStorePath;
     private final File rootDirectory;
+    private final Function<File, Long> fsSizer;
 
     public FSGitRepoStore(String repoStorePath) {
+        this(repoStorePath, d -> d.getTotalSpace() - d.getFreeSpace());
```

Comment:
The call uses whatever partition the given File lives on.

From the Java 8 docs (https://docs.oracle.com/javase/8/docs/api/java/io/File.html):

_Instances of this class may or may not denote an actual file-system object such as a file or a directory. If it does denote such an object then that object resides in a partition._

`getTotalSpace()`
_The size, in bytes, of the partition or 0L if this abstract pathname does not name a partition_

`getFreeSpace()`
_The number of unallocated bytes on the partition or 0L if the abstract pathname does not name a partition_

I guess a file "naming" a partition is ambiguous, so I've manually checked that we've got the correct behaviour on Ubuntu 14.04.5 LTS and macOS 10.12.

If it's wrong, the calls will return 0 and the tests (specifically, `FSGitRepoStoreTest.totalSizeShouldChangeWhenFilesAreCreatedAndDeleted()`) will fail.